src/WebUI/Common/KubeSummary.tsx

```diff
@@ -232,6 +232,15 @@ export class KubeSummary extends React.Component<IKubeSummaryProps, IKubernetesC
         }, 100);
     }
 
+    public componentDidUpdate(prevProps: IKubeSummaryProps, prevState: IKubernetesContainerState) {
```

Comment:
this will be called even when the workload itself is not called. when ever setstate is set, this function is called. i would want to call this when _setNamespaceOnDeploymentsFetched is called, otherwise we may be calling getservices unnecessarly

Reply:
Updated the PR - added another event zeroDeploymentsFound. KubeSummary listens to it and fetches services only once if event is triggered.